plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java

```diff
@@ -377,6 +377,14 @@ public void onCompassAccuracyChange(int compassStatus) {
     // Currently don't handle this inside SDK
   }
 
+  private void addCameraListener() {
+    if (locationLayerMode == LocationLayerMode.NAVIGATION) {
+      mapboxMap.removeOnCameraMoveListener(this);
+      return;
```

Comment:
it's a bit strange that a method called #addCameraListener is actually removing the listener in some cases. Either rename the method to something fitting or decouple the removal into a separate codeblock